최근 미팅정보 조회 API

HyunsangHan opened the pull request:
- 여러 미팅 건 중에서, 서비스상에는 '현재 진행중'인 미팅 건을 골라내서 단 하나의 미팅만 클라이언트에 노출해줄 필요가 있음. 그를 위해 주로 GET방식으로 쓰이게 될 API임
- Meeting.objects.all()의 쿼리셋에서 '아직 미팅약속일자가 되지 않은 것들 중 가장 늦은 건(현재 기준으로는 가장 가까운 미래의 건)을 뽑는다'는 방식으로 구현했음. 하단의 코드 참고.
> queryset = Meeting.objects.filter(meeting_time__gte=timezone.now()).order_by('meeting_time').first()

- 추가로, 그외에 마이너한 수정이 있었음.(문자열포맷팅부분)

HyunsangHan committed:
현재 서비스상에 노출되어야 할 미팅 건(current_meeting)에 대한 API 개발

jaewooMaeng commented on backend/backend/urls.py:
```diff
@@ -23,5 +23,6 @@
     path('admin/', admin.site.urls),
-    path('api/', include(router.urls))
+    path('api/', include(router.urls)),
+    path('api/current_meeting/', views.CurrentMeeting.as_view())
```
url에 api/ 는 굳이 안 붙여도 괜찮지 않을까요!? 사실 모든 요청은 api라고 볼 수 있어서 없어도 된다고 생각합니다. front에서 요청 없이 link로 다른 url의 컴포넌트를 보여줄 수 있긴하지만(우리 프로젝트에서는 그것도 거의 없을 듯?) 그것과 굳이 구분하기 위해 api를 붙일 필요는 없다고 생각합니다!

HyunsangHan replied:
@qwerty98765 맞는 말이네요. api/는 제외해도 될 것 같습니다.
아마 'api/'를 '/'로 수정하면 되겠죠?

HyunsangHan replied:
알아보니, '/'가 아니라 ''를 넣어야 하는 듯합니다. 수정하겠습니다.

jaewooMaeng commented on backend/meeting/serializers.py:
```diff
@@ -2,6 +2,11 @@
 class MeetingSerializer(serializers.ModelSerializer):
+    class Meta:
+        model = Meeting
+        fields = '__all__'
```
나중에 field들이 더 확실해지면 fields를 '__all__' 말고 하나하나 써주는 게 더 좋을까요? 아니면 놔두는 편이 좋을까요?

HyunsangHan replied:
@qwerty98765 아무래도 딱 필요한 fields만 골라서 써주면 좋긴 할 것 같네요.
이것도 우선 나중에 프론트개발할 때 어떻게 필요할지 모르니 일단 all로 남겨둬보는 걸로 합시다!

jaewooMaeng commented on backend/meeting/views.py:
```diff
@@ -1,9 +1,27 @@
-from rest_framework import viewsets
-from .serializers import MeetingSerializer
+from rest_framework import viewsets, status
+from django.http import Http404
```
혹시 Http404만 따로 import하는 이유는 무엇인가요!

HyunSangHan replied:
@qwerty98765 아 이게 이거 다음 API 개발할 때 예외처리하려고 import해둔 거였는데 이 API랑 관계없이 여기에 남아있었네요. 지워도 상관없긴 한데 곧 또 import해오게 될 것 같아서 일단 냅둬보고 나중에 불필요하면 지우겠습니다~

HyunsangHan replied:
아, 따로 import할 필요 없을 것 같아요~ 지우겠습니다

jaewooMaeng commented on backend/meeting/views.py:
```diff
@@ -1,9 +1,27 @@
+class CurrentMeeting(APIView):
+    def get(self, request, format=None):
+        # 미팅일자가 현재보다 미래인 경우 + 가장 빨리 디가오는 미팅 순으로 정렬해서 + 가장 앞에 있는 미팅일정 1개만 쿼리셋에 담기 
+        queryset = Meeting.objects.filter(meeting_time__gte=timezone.now()).order_by('meeting_time').first()
```
__gte에 해당되는 instance가 없는 경우에 대해서도 생각을 해도 좋을 것 같아요!

HyunsangHan replied:
@qwerty98765 instance가 없을 때의 예외처리 말이지요? 우선 get이 아닌 filter이기 때문에 에러는 안날 텐데,
제 생각엔 instance가 없다면 지금처럼 api는 빈 쿼리셋을 response해주되,
빈쿼리셋이 날라왔을 때 프론트단에서 조건문으로 처리하면 어떨지 싶네요!

HyunsangHan replied:
일단 API내에도 조건문을 넣었습니다!

HyunsangHan committed:
urls.py urlpatterns에서 include 기본경로 api/ 제거(불필요)

HyunsangHan committed:
urls.py urlpatterns에서 include 기본경로 '/'에서 ''로 수정

HyunsangHan committed:
views.py에서 http404 import 삭제

HyunsangHan committed:
views.py GET방식 http404에서 serializer.errors 삭제

UnboundLocalError: local variable 'serializer' referenced before assignment 에러가 뜨는 것을 방지하기 위해 삭제했음
